wikimedia/wikipedia-ios
# Image recommendations: Add onboarding modal

**Phabricator:** https://phabricator.wikimedia.org/T358896

### Notes
This PR adds the images recommendations onboarding modal. It also updates the `WKOnboardViewCell` to be able to support filling the background state of an icon if required by design. I also added a `WKData` data controller for image recs (similar to how Watchlist was patterned), but if this isn't gonna be our desired way to read/manipulate data (like the user defaults here) we can figure something else out.

### Test Steps
1. Optionally comment out `guard !dataController.hasPresentedOnboardingModal else ...` for repeated presentation of the modal
2. Run on Staging
3. Tap the Add button on the explore feed card
4. Confirm the modal is displayed
